metadata-ingestion/src/datahub/emitter/rest_emitter.py

```diff
@@ -338,8 +346,15 @@ def emit_usage(self, usageStats: UsageAggregation) -> None:
 
     def _emit_generic(self, url: str, payload: str) -> None:
         curl_command = make_curl_command(self._session, "POST", url, payload)
+        payload_size = len(payload)
+        if payload_size > INGEST_MAX_PAYLOAD_BYTES:
+            # since we know total payload size here, we could simply avoid sending such payload at all and report a warning, with current approach we are going to cause whole ingestion to fail
```

Comment:
Hey, I’m a bit confused about this comment.
If we send a payload that's too big, it’ll fail on the backend only and no impact in the ingestion pipeline, right?
this is how it works both before and with the changes in this PR, correct?

Reply:
The ingestion will be marked as failed if the payload size is exceeded (GMS will return 400).

Reply:
Got it!

The backend may better respond with `413 Content Too Large` and then ingestor may easily skip https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/413

Reply:
Agreed. Whether to skip such error though - that's another question.